cpu/stm32/periph/spi.c

```diff
@@ -64,6 +68,18 @@ static inline bool _use_dma(const spi_conf_t *conf)
 }
 #endif
 
+static uint8_t _get_clkdiv(const spi_conf_t *conf, uint32_t clock)
+{
+    uint32_t bus_clock = periph_apb_clk(conf->apbbus);
+    for (unsigned i = 0; i <= BR_MAX; i++) {
+        clock <<= 1;
```

Comment:
How about `bitarithm_msb(bus_clock / clock)`

Reply:
I haven't measured it, but I expect this to be slower on the stm32f0 series. I can give it a shot on the f1 and f4 if you like :)

Reply:
On the f1 (cortex-m3) this results in a constant 5.86 microsecond. for acquire/release, or an increase of approx 450 nanoseconds.

Reply:
See also the latest fixup.